receiver/podmanreceiver/receiver.go

```diff
@@ -87,3 +168,56 @@ func (r *receiver) scrape(context.Context) (pdata.Metrics, error) {
 	}
 	return md, nil
 }
+
+func (r *receiver) handleEvents(ctx context.Context, eventBackoff *backoff.ExponentialBackOff) error {
+	c, err := r.clientFactory(r.set.Logger, r.config)
+	if err != nil {
+		r.set.Logger.Error("error fetching/processing events", zap.Error(err))
+		return err
+	}
+	r.client = c
+
+	// Fetch the response from the endpoint
+	response, err := r.client.getEventsResponse(ctx)
+	if err != nil {
+		r.set.Logger.Error("error while fetching events", zap.Error(err))
+		return err
+	}
+
+	dec := json.NewDecoder(response.Body)
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			// Translate the response to the events format.
+			decodedEvent, err := decodeEvents(dec)
+			if err != nil {
```

Comment:
It can be EOF right? Should not return in that case

Reply:
Done